File: backend/internal/model/pokemon.go
package model

// FlavorTextPair holds a pair of EN/JA descriptions from the same game version(s).
type FlavorTextPair struct {
	VersionNames  []string `json:"version_names"`
	DescriptionEN string   `json:"description_en"`
	DescriptionJA string   `json:"description_ja"`
}

// Pokemon represents a Pokemon's core data fetched from PokeAPI.
type Pokemon struct {
	ID            int              `json:"id"`
	NameEN        string           `json:"name_en"`
	NameJA        string           `json:"name_ja"`
	DescriptionEN string           `json:"description_en"`
	DescriptionJA string           `json:"description_ja"`
	SpriteURL     string           `json:"sprite_url"`
	BaseStatTotal int              `json:"base_stat_total"`
	Types         []string         `json:"types"`
	Height        int              `json:"height"`
	Weight        int              `json:"weight"`
	FlavorTexts   []FlavorTextPair `json:"flavor_texts,omitempty"`
}

File: backend/src/types/pokemon.ts
/** ポケモン図鑑説明文の英日ペア。複数バージョンの説明文を 1 件にまとめる。 */
export interface FlavorTextPair {
  version_names: string[];
  description_en: string;
  description_ja: string;
}

/** PokeAPI から取得したポケモン情報を内部表現に変換した型。 */
export interface Pokemon {
  id: number;
  name_en: string;
  name_ja: string;
  description_en: string;
  description_ja: string;
  sprite_url: string;
  base_stat_total: number;
  types: string[];
  height: number;
  weight: number;
  is_legendary: boolean;
  is_mythical: boolean;
  flavor_texts?: FlavorTextPair[];
}
